### ISSUE TITLE
self.sock.recv() causes hang

### ISSUE THREAD

**Issue by martinezjulio** [alters-mit/magnebot#8]
I am trying to install using pip on my Mac. After running the controller I always get a hang after ```m=Controller()```. The imports seems to work fine however. I hit control+C to quit and always get the following output which seems like the self.socket.recv() line is making the program hang for me:

```
line 50, in __init__
    self.socket.recv()
  File "zmq/backend/cython/socket.pyx", line 781, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 817, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 13, in zmq.backend.cython.checkrc._check_rc
```

Any help would be appreciated. Thank you.

**Comment by alters-mit** [comment]
@martinezjulio 

For the Magnebot API, you should probably be doing this:

```python
from magnebot import Magnebot

m = Magnebot()
```

Regardless of whether you instantiate a `Controller()` or a `Magnebot()`, you should get this message:

```
Your installed tdw Python module is up to date with PyPi.
```

Do you see that message printed? If not, there might be an issue with your Internet connection (the controller is trying to get the most recent version of TDW from [here](https://pypi.org/project/tdw/)).

**Comment by martinezjulio** [comment]
@alters-mit 

Yes. I am importing that exact way as specified on your example controller and I do see the same message printed on my terminal. However, right after that message is where the program hangs until I hit control+c.

**Comment by alters-mit** [comment]
This controller should automatically launch the build:

```python
from tdw.controller import Controller
c = Controller()
print("OK!")
c.communicate({"$type": "terminate"})
```

Does it print `OK!` ?

***

This controller  will _not_ automatically launch the build. You need to launch the build manually (there is a [known issue](https://github.com/threedworld-mit/tdw/issues/159) with double-clicking the .app; for now, you can run TDW.app/Content/MacOS/tdw in the terminal)

```python
from magnebot import Magnebot
m = Magnebot()
print("OK!")
m.end()
```

Does it print `OK!` ?

**Comment by martinezjulio** [comment]
For me it is saved with capitals as TDW/TDW.app/Contents/MacOS/TDW. When I run that in the terminal however I get a pop up prompt that says: “TDW” is damaged and can’t be opened. Given that the program till hangs at m=Magnebot(). I tried reinstalling and re-downloading tdw but I get the same error.

**Comment by martinezjulio** [comment]
Btw I should add that I am able to run the test controllers from the getting started page for tdw just fine. So I am guessing controllers using Controller from tdw does automatically launch the build?

So for instance running this example works fine.
```
from tdw.controller import Controller

c = Controller()
print("Everything is OK!")
c.communicate({"$type": "terminate"})
```

**Comment by alters-mit** [comment]
Huh! I knew that double-clicking the .app would cause a warning that "TDW" is damaged. I'll fix this soon... (The bottleneck is that I don't actually own a Mac)

Does this work?

```python
from magnebot import Magnebot
m = Magnebot(launch_build=True)
print("OK!")
m.end()
```

> So I am guessing controllers using Controller from tdw does automatically launch the build?

Yes. In `Controller`, the default value for `launch_build` is True and in `Magnebot` the default value of `launch_build` is False. The reasoning behind that is mostly due to TDW being written well before Magnebot, and that I didn't expect someone would want to use both classes! In either case, `launch_build=True` will automatically launch the build except on Linux servers, in which case you should always set `launch_build=False`.

**Comment by martinezjulio** [comment]
Ok perfect that works! Thank you!!

**Comment by alters-mit** [comment]
Fixed in TDW 1.8.9;  TDW.app should no longer have "damaged" errors.
To update: `pip3 install tdw -U`.

- If you're using the Magnebot API: Download the latest build of TDW. When you extract the zip file, you'll see `setup.sh` in the same folder as `TDW.app`. Run it and you should be able to use `TDW.app` normally.
- If you're using TDW:  The next time you call `c = Controller()` the controller will automatically download and repair `TDW.app`
